core/src/main/java/io/questdb/cutlass/text/TextMetadataParser.java

```diff
@@ -194,16 +199,17 @@ private void clearStage() {
         locale = null;
         localePosition = 0;
         utf8 = false;
+        index = false;
     }
 
     private CharSequence copy(CharSequence tag) {
         final int l = tag.length() * 2;
         final long n = bufSize + l;
         if (n > bufCapacity) {
-            long ptr = Unsafe.malloc(n * 2);
-            Unsafe.getUnsafe().copyMemory(buf, ptr, bufSize);
+            long ptr = Unsafe.malloc(n * 2, MemoryTag.NATIVE_DEFAULT);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).